sumpy/tools.py

```diff
@@ -976,52 +983,262 @@ def wait(self):
+            f"[ifft_{ilev}]: 0<=ifft_{ilev}<{nfft}",
+            f"[iN1_{ilev}]: 0<=iN1_{ilev}<{N1}",
+            f"[iN1_sum_{ilev}]: 0<=iN1_sum_{ilev}<{N1}",
+            f"[iN2_{ilev}]: 0<=iN2_{ilev}<{N2}",
+            f"[i_{ilev}]: 0<=i_{ilev}<{n}",
+            f"[i2_{ilev}]: 0<=i2_{ilev}<{n}",
+        ]
+
+    for idom, dom in enumerate(domains):
+        if not dom.startswith("{"):
+            domains[idom] = "{" + dom + "}"
+
+    kernel_data = [
+        lp.GlobalArg("x", shape=shape, is_input=False, is_output=True,
+            dtype=complex_dtype),
+        lp.GlobalArg("y", shape=shape, is_input=True, is_output=False,
+            dtype=complex_dtype),
+        lp.TemporaryVariable("exp_table", shape=(n,),
+            dtype=complex_dtype),
+        lp.TemporaryVariable("temp", shape=(n,),
+            dtype=complex_dtype),
+        ...
+    ]
+
+    if n == 1:
+        domains = domains[2:]
+        insns = [
+            lp.Assignment(
+                assignee=x[(*broadcast_dims, 0)],
+                expression=y[(*broadcast_dims, 0)],
+            ),
+        ]
+        kernel_data = kernel_data[:2]
+    elif inverse:
+        domains += ["{[i3]: 0<=i3<n}"]
+        insns += [
+            lp.Assignment(
+                assignee=x[(*broadcast_dims, i3)],
+                expression=x[(*broadcast_dims, i3)] / n,
+                depends_on=frozenset([f"update_{len(factors) - 1}"]),
+            ),
+        ]
+
+    if name is None:
+        if inverse:
+            name = f"ifft_{n}"
+        else:
+            name = f"fft_{n}"
+
+    knl = lp.make_kernel(
+        domains, insns,
+        kernel_data=kernel_data,
+        name=name,
+        fixed_parameters=fixed_parameters,
+        lang_version=lp.MOST_RECENT_LANGUAGE_VERSION,
+        index_dtype=index_dtype,
+    )
+
+    if broadcast_dims:
+        knl = lp.split_iname(knl, "j0", 32, inner_tag="l.0", outer_tag="g.0")
+        knl = lp.add_inames_for_unused_hw_axes(knl)
+
+    return knl
+
+
+class FFTBackend(enum.Enum):
+    pyvkfft = 1
+    loopy = 2
+
+
+def _get_fft_backend(queue) -> FFTBackend:
+    env_val = os.environ.get("SUMPY_FFT_BACKEND", None)
+    if env_val:
+        if env_val not in ["loopy", "pyvkfft"]:
+            raise ValueError("Expected 'loopy' or 'pyvkfft' for SUMPY_FFT_BACKEND. "
+                   f"Found {env_val}.")
+        return FFTBackend[env_val]
+
+    try:
+        import pyvkfft.opencl  # noqa: F401
+    except ImportError:
+        warnings.warn("VkFFT not found. FFT runs will be slower.")
+        return FFTBackend.loopy
+
     if queue.properties & cl.command_queue_properties.OUT_OF_ORDER_EXEC_MODE_ENABLE:
-        raise RuntimeError("VkFFT does not support out of order queues yet.")
+        warnings.warn("VkFFT does not support out of order queues yet. "
+            "Falling back to slower implementation.")
+        return FFTBackend.loopy
+
+    if (sys.platform == "darwin"
+            and platform.machine == "x86_64"
```

Comment:
`platform.machine` is a function I think.

Reply:
```suggestion
            and platform.machine() == "x86_64"
```